extension/source/components/Button.tsx

```diff
@@ -1,21 +1,37 @@
 import { IconProps } from 'phosphor-react';
-import { ReactElement, ReactNode } from 'react';
+import { ReactElement, ReactNode, useState } from 'react';
 
 const Button = (props: {
   className?: string;
-  onPress: () => void;
+  onPress: () => unknown;
   loading?: boolean;
   children?: ReactNode;
   icon?: IconProps;
 }): ReactElement => {
+  const [pressLoading, setPressLoading] = useState(false);
+
   return (
     <div
       className={
-        props.loading
+        props.loading || pressLoading
           ? 'btn-loading'
           : `flex gap-2 items-center select-none ${props.className}`
       }
-      onClick={props.onPress}
+      onClick={async () => {
+        const pressResult = props.onPress();
+
+        if (pressResult instanceof Promise) {
+          setPressLoading(true);
+
+          try {
+            await pressResult;
+          } catch {
+            // Enhancement: Give the button an error state
```

Comment:
I think this will swallow the error, would be good to log.

```suggestion
          } catch(err) {
            // Enhancement: Give the button an error state
            console.error(err);
```